desktop/cmp/persistenceManager/PersistenceManagerModel.ts

```diff
@@ -0,0 +1,345 @@
+    /** Used to persist this model's selected ID. */
+    persistWith: PersistOptions;
+    /** True (default) to render a save button alongside the primary menu button when dirty. */
+    enableTopLevelSaveButton?: boolean;
+    /** Fn to produce a new, empty object - can be async. */
+    newObjectFnAsync?: () => PlainObject;
+}
+
+export class PersistenceManagerModel extends HoistModel {
+    //------------------------
+    // Persistence Provider
+    // Pass this to models that implement `persistWith` to include their state in the view.
+    //------------------------
+    provider: PersistOptions = {
+        getData: () => cloneDeep(this.pendingValue ?? this.value ?? {}),
+        setData: value => this.mergePendingValue(value)
+    };
+
+    @observable.ref @managed manageDialogModel: ManageDialogModel;
+
+    @observable.ref @managed saveDialogModel: SaveDialogModel;
+
+    enableTopLevelSaveButton: boolean = true;
+
+    newObjectFn: () => PlainObject;
+
+    /** Current state of the active object, can include not-yet-persisted changes. */
+    @observable.ref pendingValue: PlainObject = null;
+
+    @observable.ref views: PlainObject[] = [];
+
+    @bindable selectedId: StoreRecordId;
+
+    /** Reference Key used to query JsonBlobs */
+    type: string;
+
+    /** Configured word to label an object persisted by this model. */
+    noun: string;
+
+    get pluralNoun(): string {
+        return pluralize(this.noun);
+    }
+
+    get capitalNoun(): string {
+        return capitalize(this.noun);
+    }
+
+    get capitalPluralNoun(): string {
+        return capitalize(this.pluralNoun);
+    }
+
+    private readonly canManageGlobalFn: Thunkable<boolean>;
+
+    get canManageGlobal(): boolean {
+        return executeIfFunction(this.canManageGlobalFn);
+    }
+
+    onChangeAsync?: (value: PlainObject) => void;
+
+    get value(): PlainObject {
+        return this.selectedObject?.value;
+    }
+
+    get objects(): PlainObject[] {
+        return this.views;
+    }
+
+    get selectedObject(): PlainObject {
+        return this.views.find(it => it.id === this.selectedId);
+    }
+
+    @computed
+    get canSave(): boolean {
+        const {value} = this;
+        if (!value) return false;
+        return (
+            this.isDirty && (this.canManageGlobal || value.isShared) && !this.loadModel.isPending
+        );
+    }
+
+    @computed
+    get isDirty(): boolean {
+        return !this.isEqualSkipAutosize(this.pendingValue, this.value);
+    }
+
+    get isShared(): boolean {
+        return !!this.selectedObject?.isShared;
+    }
+
+    // Internal persistence provider, used to save *this* model's state, i.e. selectedId
+    _provider;
+
+    constructor({
+        type,
+        noun,
+        onChangeAsync,
+        persistWith,
+        canManageGlobal,
+        enableTopLevelSaveButton = true,
+        newObjectFnAsync
+    }: PersistenceManagerProps) {
+        super();
+        makeObservable(this);
+
+        throwIf(
```

Comment:
I would take this out. It's clear from the type that `onChangeAsync` is a required property.